_posts/en/newsletters/2025-10-31-newsletter.md

```diff
@@ -0,0 +1,163 @@
+
+- [Eclair 0.13.1][] is a minor release of this LN node implementation.  This
+  release contains database changes to prepare for the removal of pre-[anchor
+  output][topic anchor outputs] channels. You will need to run the v0.13.0
+  release first to migrate your channel data to the latest internal encoding.
+
+## Notable code and documentation changes
+
+_Notable recent changes in [Bitcoin Core][bitcoin core repo], [Core
+Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
+[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
+Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
+Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
+Proposals (BIPs)][bips repo], [Lightning BOLTs][bolts repo],
+[Lightning BLIPs][blips repo], [Bitcoin Inquisition][bitcoin inquisition
+repo], and [BINANAs][binana repo]._
+
+- [Bitcoin Core #29640][] reinitializes `nSequenceId` values at startup for
+  blocks on competing chains with the same amount of work: 0 for blocks
+  belonging to the previously known best chain, and 1 for all other loaded
+  blocks. This resolves an issue where Bitcoin Core couldn’t perform a tiebreak
+  between two competing chains because the `nSequenceId` value didn't persist
+  across restarts.
+
+- [Core Lightning #8400][] introduces a new [BIP39][] mnemonic backup format for
+  the `hsm_secret` with optional passphrase for all new nodes by default, while
+  keeping support for legacy 32-byte `hsm_secrets` on existing nodes. `Hsmtool`
+  is also updated to support both mnemonic-based and legacy secrets. A new
+  standard [Taproot][topic taproot] derivation format is introduced for wallets.
+
+- [Eclair #3173][] removes support for legacy channels that don’t use [anchor
+  outputs][topic anchor outputs] or [taproot][topic taproot], also known as
+  `static_remotekey` or `default` channels. Users should close any remaining
+  legacy channels before upgrading to version 0.13 or 0.13.1.
+
+- [LND #10280][] now waits for the headers to sync before starting the chain
+  notifier (see [Newsletter #31][news31 chain]) to rescan the chain for wallet
+  transactions. This resolves an issue in which LND would trigger a premature
+  rescan while the headers were still syncing when a new wallet was created.
+  This primarily affected [Neutrino backends][topic compact block filters].
+
+- [LDK #5516][] updates the responses of `compute_merkle_root` and
+  `compute_witness_root` to return `None` when the transaction list contains
+  adjacent duplicates. This prevents the mutated [Merkle root
+  vulnerability][topic merkle tree vulnerabilities] CVE 2012-2459, where an
+  invalid block with a duplicated transaction can share the same Merkle root
+  (and blockhash) as a valid block, leading LDK to confuse and reject both. This
+  solution is inspired by Bitcoin Core’s.
+
+- [BIPs #2006][] updates [BIP3][]’s specification (see [Newsletter #344][news344
```

Comment:
```suggestion
- [BIP #2006][] updates [BIP3][]’s specification (see [Newsletter #344][news344
```

Reply:
BIPs is how we refer to the BIPs repository historically